error TS2307: Cannot find module '@sentry/node/build/types/integrations/anr/common' or its corresponding type declarations.

### Is there an existing issue for this?

- [x] I have checked for existing issues https://github.com/getsentry/sentry-javascript/issues
- [x] I have reviewed the documentation https://docs.sentry.io/
- [x] I am using the latest SDK release https://github.com/getsentry/sentry-javascript/releases

### How do you use Sentry?

Sentry Saas (sentry.io)

### Electron SDK Version

5.3.0

### Electron Version

31.3.1

### What platform are you using?

Linux

### Link to Sentry event

_No response_

### Steps to Reproduce

```shell
npm i @sentry/electron electron typescript
echo 'import * as Sentry from "@sentry/electron/main"' > test.ts
npx tsc --module esnext --moduleResolution bundler test.ts
```

### Expected Result

No errors.

### Actual Result

```
node_modules/@sentry/electron/esm/main/integrations/anr.d.ts:4:64 - error TS2307: Cannot find module '@sentry/node/build/types/integrations/anr/common' or its corresponding type declarations.

4 export declare const anrIntegration: (options?: Partial<import("@sentry/node/build/types/integrations/anr/common").AnrIntegrationOptions> | undefined) => import("@sentry/types").Integration;
                                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error in node_modules/@sentry/electron/esm/main/integrations/anr.d.ts:4
```

This is a regression in 5.0.0-beta.0.

--- a/package.json
+++ b/package.json
@@ -39,7 +39,7 @@
   "scripts": {
     "prebuild": "yarn clean && node scripts/update-version.js",
     "build": "rollup --config rollup.config.mjs",
-    "postbuild": "node scripts/check-exports.mjs",
+    "postbuild": "node scripts/check-exports.mjs && node scripts/check-types.mjs",
     "clean": "rimraf --glob coverage common esm main preload renderer index.* sentry-electron*.tgz .eslintcache",
     "prelint": "node scripts/update-version.js",
     "lint": "run-s lint:prettier lint:eslint",

--- a/scripts/check-types.mjs
+++ b/scripts/check-types.mjs
@@ -1,0 +1,39 @@
+import { promises as fs } from 'node:fs';
+import * as path from 'node:path';
+
+async function walk(dir, ty) {
+  let files = await fs.readdir(dir);
+  files = await Promise.all(
+    files.map(async (file) => {
+      const filePath = path.join(dir, file);
+      const stats = await fs.stat(filePath);
+      if (stats.isDirectory()) {
+        return filePath.endsWith('node_modules') ? [] : walk(filePath, ty);
+      } else if (stats.isFile() && file.endsWith(ty)) {
+        return filePath;
+      }
+    }),
+  );
+
+  return files.filter(Boolean).reduce((all, f) => all.concat(f), []);
+}
+
+// Find all the .d.ts files in the project
+const typeDefs = await walk(process.cwd(), '.d.ts');
+
+// Check for any sub-module imports in the type definitions
+const regex = /import\("@sentry\/.+\//gi;
+let failed = false;
+
+for (const file of typeDefs) {
+  const content = await fs.readFile(file, 'utf-8');
+  const matches = content.match(regex);
+  if (matches) {
+    console.log(`Sub-module import in type def file '${file}':\n`, matches);
+    failed = true;
+  }
+}
+
+if (failed) {
+  process.exit(1);
+}

--- a/src/main/integrations/anr.ts
+++ b/src/main/integrations/anr.ts
@@ -1,13 +1,16 @@
 import { defineIntegration } from '@sentry/core';
 import { anrIntegration as nodeAnrIntegration } from '@sentry/node';
+import { Integration } from '@sentry/types';
 import { app, powerMonitor } from 'electron';
 
 import { ELECTRON_MAJOR_VERSION } from '../electron-normalize';
 
+type Options = Parameters<typeof nodeAnrIntegration>[0];
+
 /**
  * Starts a worker thread to detect App Not Responding (ANR) events
  */
-export const anrIntegration = defineIntegration((options: Parameters<typeof nodeAnrIntegration>[0] = {}) => {
+export const anrIntegration: (options: Options) => Integration = defineIntegration((options: Options = {}) => {
   if (ELECTRON_MAJOR_VERSION < 22) {
     throw new Error('Main process ANR detection requires Electron v22+');
   }

